mod/state-transition/pkg/core/state_processor_genesis.go

```diff
@@ -47,6 +47,11 @@ func (sp *StateProcessor[
 	executionPayloadHeader ExecutionPayloadHeaderT,
 	genesisVersion common.Version,
 ) (transition.ValidatorUpdates, error) {
+	sp.processingGenesis = true
+	defer func() {
+		sp.processingGenesis = false
+	}()
```

Comment:
we are sure there is only 1 thread ever calling into this?

Reply:
Yes as long as we use CometBFT as consensus.